pkg/k8shandler/fluentd.go

```diff
@@ -153,14 +154,36 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateFluentdServiceMonitor
 		},
 	}
 
-	utils.AddOwnerRefToObject(serviceMonitor, utils.AsOwner(cluster))
+	utils.AddOwnerRefToObject(desired, utils.AsOwner(cluster))
 
-	err := clusterRequest.Create(serviceMonitor)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("Failure creating the fluentd ServiceMonitor: %v", err)
+	err := clusterRequest.Create(desired)
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("Failure creating the fluentd ServiceMonitor: %v", err)
+		}
+		current := &monitoringv1.ServiceMonitor{}
+		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			if err = clusterRequest.Get(desired.Name, current); err != nil {
+				if errors.IsNotFound(err) {
+					// the object doesn't exist -- it was likely culled
+					// recreate it on the next time through if necessary
+					return nil
+				}
+				return fmt.Errorf("Failed to get %q service for %q: %v", current.Name, clusterRequest.Cluster.Name, err)
+			}
+			if servicemonitor.AreSame(current, desired) {
+				log.V(3).Info("ServiceMonitor are the same skipping update")
+				return nil
+			}
+			//Explicitly copying because services are immutable
+			current.Labels = desired.Labels
```

Comment:
We should also set the annotations since the comparison relies upon that as well, or remove the annotation comparison

Reply:
ok